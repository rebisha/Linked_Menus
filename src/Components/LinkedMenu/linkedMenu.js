import React, { Fragment, useState } from "react";
// component
import Button from "../Button/button";
import Form from "../Form/form";

const LinkedMenu = ({ subMenu, handleCancel }) => {
  const [tertiaryMenuToggle, setTertiaryMenuToggle] = useState(false);
  const [toggleForm, setToggleForm] = useState(false);

  const toggleTertiary = () => {
    setTertiaryMenuToggle(!tertiaryMenuToggle);
  };

  const formToggle = () => {
    setToggleForm(!toggleForm);
  };

  const edit = editIndex => {
    setEditIndex(editIndex);
  };

  const showNestedMenu = item => {
    const { id, name, subMenu, hasForm } = item;

    if (!subMenu) {
      return (
        <Fragment>
          <Button
            type="button"
            className="tertiary-menu"
            key={id}
            name={name}
            onClick={formToggle}
          />
          {hasForm && toggleForm && <Form handleCancel={formToggle} />}
          {!hasForm && toggleForm && (
            <h1>This functionality hasn't been implemented yet!!</h1>
          )}
        </Fragment>
      );
    } else {
      return (
        <Fragment>
          <Button
            type="button"
            className="tertiary-menu"
            key={id}
            name={name}
            onClick={toggleTertiary}
          />
          <div className="toggle-option ">
            {subMenu &&
              tertiaryMenuToggle &&
              subMenu.map(item => (
                <Button
                  type="button"
                  className="tertiary-menu"
                  key={item.id}
                  name={item.name}
                />
              ))}
          </div>
        </Fragment>
      );
    }
  };

  return <div className="toggle-option">{showNestedMenu(subMenu)}</div>;
};

export default LinkedMenu;
