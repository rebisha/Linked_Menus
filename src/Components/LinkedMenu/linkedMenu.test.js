import React from "react";
import LinkedMenu from "./linkedMenu";
import renderer from "react-test-renderer";

const subMenu = [
  {
    name: "Example 1",
    hasForm: false,
    id: "examplekey0"
  },
  {
    name: "Example 2",
    hasForm: false,
    id: "examplekey1"
  },
  {
    name: "Example 3",
    hasForm: true,
    id: "examplekey2",
    subMenu: [
      {
        id: 1,
        name: "SubMenu 1"
      },
      {
        id: 1,
        name: "SubMenu 1"
      }
    ]
  }
];

test("LinkedMenu component renders correctly", () => {
  const component = renderer.create(
    <LinkedMenu handleCancel={() => !subMenuToggle} subMenu={subMenu} />
  );
  expect(component).toMatchSnapshot();
});
