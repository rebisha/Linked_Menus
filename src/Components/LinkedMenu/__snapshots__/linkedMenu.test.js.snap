// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkedMenu component renders correctly 1`] = `
<div
  className="toggle-option"
>
  <button
    className="button-wrapper"
    onClick={[Function]}
    type="button"
  />
</div>
`;
